libraries/chain/host_context.cpp

```diff
@@ -75,6 +75,11 @@ uint32_t host_context::execute_sync_call(name call_receiver, uint64_t flags, std
          try {
             // use a new sync_call_context for next sync call
             sync_call_context call_ctx(control, trx_context, get_sync_call_sender(), call_receiver, depth, flags, data);
+
+            if (!control.skip_trx_checks()) {
+               call_ctx.privileged = receiver_account->is_privileged();
+            }
```

Comment:
Why not just always set `call_ctx.privileged` ?

Reply:
`exec_one()` of apply_context has this check for action execution. I did the same for consistency. Probably we can always set `call_ctx.privileged`.

Reply:
I think it about more, and have changed to always set it at sync_call_context constructor. Thanks.

https://github.com/AntelopeIO/spring/pull/1311/commits/2500086af7d096a0f9004c52bea9918679b36e28